controller/controller_server/csi_controller_server.py

```diff
@@ -454,11 +454,15 @@ def CreateSnapshot(self, request, context):
                 res = utils.generate_csi_create_snapshot_response(snapshot, source_volume_id)
                 logger.info("finished create snapshot")
                 return res
-        except (controller_errors.IllegalObjectName, controller_errors.VolumeNotFoundError,
+        except (controller_errors.IllegalObjectName,
                 controller_errors.PoolParameterIsMissing) as ex:
             context.set_details(ex.message)
             context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
             return csi_pb2.CreateSnapshotResponse()
+        except controller_errors.VolumeNotFoundError as ex:
+            context.set_code(grpc.StatusCode.NOT_FOUND)
```

Comment:
errors not really match any of the errors specified in the [spec](https://github.com/container-storage-interface/spec/blob/v1.3.0/spec.md#createsnapshot-errors)

Reply:
"If the plugin is unable to complete the CreateSnapshot call successfully, it MUST return a non-ok gRPC code in the gRPC status."
do you think INTERNAL is a better match than NOT_FOUND?

Reply:
not necessarily, but we do need to implement the spec expected errors, I dont see them

Reply:
please be more specific about what you think is missing

Reply:
RESOURCE_EXHAUSTED, ABORTED and ALREADY_EXISTS are expected to be raised in the CreateSnapshot spac, only ALREADY_EXISTS is raised (from this list..)

Reply:
we already have tickets for:
ABORTED (CSI-1050)
RESOURCE_EXHAUSTED (CSI-1664)

and anyway, they are definitively not part of this ticket.
is there anything else? or can you resolve this?